kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/datatest/datatest.kt

```diff
@@ -31,20 +25,13 @@ suspend fun <T> ContainerScope.forNone(data: List<Pair<String, T>>, test: suspen
    }
 }
 
-suspend fun <T : Any> ContainerScope.forAll(ts: List<T>, test: suspend (T) -> Unit) {
-   val idents = Identifiers()
-   ts.forEach { t ->
-      val name = idents.stableIdentifier(t)
-      addTest(createTestName(name), false) { test(t) }
-   }
-}
+suspend fun <T : Any> ContainerScope.forAll(vararg ts: T, test: suspend (T) -> Unit) = forAll(ts.toList(), test)
 
-suspend fun <T : Any> ContainerScope.forAll(vararg ts: T, test: suspend (T) -> Unit) {
-   val idents = Identifiers()
-   ts.forEach { t ->
-      val name = idents.stableIdentifier(t)
-      addTest(createTestName(name), false) { test(t) }
-   }
+suspend fun <T : Any> ContainerScope.forAll(vararg data: Pair<String, T>, test: suspend (T) -> Unit) = forAll(data.toList(), test)
```

Comment:
Tomorrow morning I need to verify that this won't get confused with the `List<T>` overload